subsys/nfc/t4t/tlv_block.c

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <string.h>
+#include <logging/log.h>
+#include <sys/byteorder.h>
+#include <nfc/t4t/tlv_block.h>
+#include <errno.h>
+
+LOG_MODULE_DECLARE(nfc_t4t_cc_file);
+
+/* Length of a type field. */
+#define TLV_TYPE_FIELD_LEN 1U
+
+/* Length of a short length field. */
+#define TLV_LEN_SHORT_FIELD_LEN 1U
+
+/* Length of an extended length field. */
+#define TLV_LEN_LONG_FIELD_LEN 3U
+
+/* Value indicating the use of an extended length field. */
+#define TLV_LEN_LONG_FORMAT_TOKEN 0xFF
+
+/* Size of long format token. */
+#define TLV_LEN_LONG_FORMAT_TOKEN_SIZE 1U
+
+/* The minimal value of length field that can be used in long format. */
+#define TLV_LEN_LONG_FORMAT_MIN_VALUE 0xFF
+
+#define TLV_MIN_TL_FIELD_LEN (TLV_TYPE_FIELD_LEN + TLV_LEN_SHORT_FIELD_LEN)
+#define TLV_MIN_LONG_FORMAT_TL_FIELD_LEN (TLV_TYPE_FIELD_LEN + TLV_LEN_LONG_FIELD_LEN)
+#define TLV_MIN_VALUE_FIELD_SIZE 6U
+
+#define FILE_CONTROL_FILE_ID_FIELD_SIZE 2U
+#define FILE_CONTROL_READ_ACCESS_FIELD_SIZE 1U
+#define FILE_CONTROL_WRITE_ACCESS_FIELD_SIZE 1U
+#define FILE_CONTROL_COMMON_FIELDS_SIZE (FILE_CONTROL_FILE_ID_FIELD_SIZE        \
+					 + FILE_CONTROL_READ_ACCESS_FIELD_SIZE  \
+					 + FILE_CONTROL_WRITE_ACCESS_FIELD_SIZE)
+
+#define FILE_ID_INVALID_VALUE_0 0x0000
+#define FILE_ID_INVALID_VALUE_1 0xE102
+#define FILE_ID_INVALID_VALUE_2 0xE103
+#define FILE_ID_INVALID_VALUE_3 0x3F00
+#define FILE_ID_INVALID_VALUE_4 0x3FFF
+#define FILE_ID_INVALID_VALUE_5 0xFFFF
+
+#define NDEF_FILE_MAX_SIZE_FIELD_SIZE 2U
+#define NDEF_FILE_MAX_SIZE_MIN_VAL 0x0005
+#define NDEF_FILE_MAX_SIZE_MAX_VAL 0xFFFE
+#define NDEF_FILE_CONTROL_TLV_LEN (FILE_CONTROL_COMMON_FIELDS_SIZE \
+				   + NDEF_FILE_MAX_SIZE_FIELD_SIZE)
+
+#define PROPRIETARY_FILE_MAX_SIZE_FIELD_SIZE 2U
+#define PROPRIETARY_FILE_MAX_SIZE_MIN_VAL 0x0003
+#define PROPRIETARY_FILE_MAX_SIZE_MAX_VAL 0xFFFE
+#define PROPRIETARY_FILE_CONTROL_TLV_LEN (FILE_CONTROL_COMMON_FIELDS_SIZE        \
+					  + PROPRIETARY_FILE_MAX_SIZE_FIELD_SIZE)
+
+#define EXTENDED_NDEF_FILE_MAX_SIZE_FIELD_SIZE 4U
+#define EXTENDED_NDEF_FILE_MAX_SIZE_MIN_VAL 0x0000FFFFUL
+#define EXTENDED_NDEF_FILE_MAX_SIZE_MAX_VAL 0xFFFFFFFEUL
+#define EXTENDED_NDEF_FILE_CONTROL_TLV_LEN (FILE_CONTROL_COMMON_FIELDS_SIZE          \
+					    + EXTENDED_NDEF_FILE_MAX_SIZE_FIELD_SIZE)
+
+/** @brief Validates maximum file size field range. This field is present
+ *         in every File Control TLV.
+ */
+static bool max_field_range_verify(u32_t value, u32_t min, u32_t max)
+{
+	return ((value > min) && (value < max));
```

Comment:
usually min and max are inclusive
e.g. this will not work if possible values start from 0

Reply:
Changed but I not sure if I understood yours comment.

Reply:
it's ok now :)

Reply:
I hope you set the min and max correctly ;)